components/style/matching.rs

```diff
@@ -379,6 +379,54 @@ pub enum StyleSharingResult {
     StyleWasShared(usize),
 }
 
+/// The result status for match primary rules.
+#[derive(Debug)]
+pub struct RulesMatchedResult {
+    /// Indicate that the rule nodes are changed.
+    rule_nodes_changed: bool,
+    /// Indicate that there are any changes of important rules overriding animations.
+    important_rules_overriding_animation_changed: bool,
+}
+
+bitflags! {
+    /// Flags that represent the result of replace_rules.
+    pub flags RulesChanged: u8 {
+        /// Normal rules are changed.
+        const NORMAL_RULES_CHANGED = 0x01,
+        /// Important rules are changed.
+        const IMPORTANT_RULES_CHANGED = 0x02,
+    }
+}
+
+impl RulesChanged {
+    /// Return true if there are any normal rules changed.
+    #[inline]
+    pub fn normal_rules_changed(&self) -> bool {
+        self.contains(NORMAL_RULES_CHANGED)
+    }
+
+    /// Return true if there are any important rules changed.
+    #[inline]
+    pub fn important_rules_changed(&self) -> bool {
+        self.contains(IMPORTANT_RULES_CHANGED)
+    }
+}
+
+/// Convert RulesMatchedResult to RulesChanged.
+impl From<RulesMatchedResult> for RulesChanged {
+    fn from(result: RulesMatchedResult) -> Self {
+        let mut rules_changed = RulesChanged::empty();
+        if result.rule_nodes_changed {
```

Comment:
Yes, per our irc talk, let's pass the boolean, instead of converting RulesMatchedResult into RulesChanged.